rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h

```diff
@@ -350,7 +350,13 @@ __attribute__((used)) void _mutex_release (OS_ID *mutex) {
 
 /* Main Thread definition */
 extern void pre_main (void);
-osThreadDef_t os_thread_def_main = {(os_pthread)pre_main, osPriorityNormal, 1U, 0U, NULL};
+
+#if defined(TARGET_MCU_NRF51822) || defined(TARGET_MCU_NRF52832)
+static uint32_t thread_stack_main[DEFAULT_STACK_SIZE / sizeof(uint32_t)];
+#else
+static uint32_t thread_stack_main[DEFAULT_STACK_SIZE * 2 / sizeof(uint32_t)];
```

Comment:
Maybe at some point it will be necessary to makes the RTOS configurable rather than trying to find the one size fits all solution:
* main stack size
* default stack size
* timers ON/OFF
* Number of concurrent threads
* etc...